File: python/plugins/solaceMessageProcessor/solace_config.py
import logging

logging.basicConfig(level=logging.DEBUG)
logger = logging.getLogger(__name__)


class SolaceConfig:
    """Configuration class for Solace connection properties."""

    def __init__(self):
        self.broker_props = {
            "solace.messaging.transport.host": "tcp://localhost:55555",
            "solace.messaging.service.vpn-name": "default",
            "solace.messaging.authentication.scheme.basic.username": "default",
            "solace.messaging.authentication.scheme.basic.password": ""
        }
        self.topic_prefix = "trade"
        self.topics = {
            "new_order": f"{self.topic_prefix}/new_order",
            "cancel": f"{self.topic_prefix}/cancel",
            "amend": f"{self.topic_prefix}/amend"
        }

File: java/plugins/solaceMessageProcessor/src/main/java/com/applicationsx/trademessageprocessor/SolaceConfig.java
package com.applicationsx.trademessageprocessor;

import java.util.HashMap;
import java.util.Map;
import java.util.Properties;

public class SolaceConfig {
    public static final String HOST = "tcp://localhost:55555";
    public static final String VPN = "default";
    public static final String USERNAME = "default";
    public static final String PASSWORD = "";

    public static final String TOPIC_PREFIX = "trade";
    public static final Map<String, String> TOPICS = new HashMap<>();

    public static final Properties brokerProps = new Properties();

    static {
        TOPICS.put("new_order", TOPIC_PREFIX + "/new_order");
        TOPICS.put("cancel", TOPIC_PREFIX + "/cancel");
        TOPICS.put("amend", TOPIC_PREFIX + "/amend");
        
        brokerProps.setProperty("solace.messaging.transport.host", HOST);
        brokerProps.setProperty("solace.messaging.service.vpn-name", VPN);
        brokerProps.setProperty("solace.messaging.authentication.scheme.basic.username", USERNAME);
        brokerProps.setProperty("solace.messaging.authentication.scheme.basic.password", PASSWORD);
    }
}
